AlphaWallet/WalletConnect/WalletConnectServer.swift

```diff
@@ -382,6 +344,53 @@ extension WalletConnectServer: ServerDelegate {
             self.removeSession(for: session.url)
         }
     }
+
+    private func addOrUpdateSession(session: Session) {
+        guard var sessions = _sessions.value else { return }
+        if let index = sessions.firstIndex(where: { $0.dAppInfo.peerId == session.dAppInfo.peerId }) {
+            sessions[index] = session
+        } else {
+            sessions.append(session)
+        }
+
+        UserDefaults.standard.walletConnectSessions = sessions
+        refresh(sessions: sessions)
+    }
+
+    func server(_ server: Server, didUpdate session: Session) {
+        debug("WalletConnect didUpdate: \(session.url.absoluteString)")
+        DispatchQueue.main.async {
+            self.addOrUpdateSession(session: session)
+        }
+    }
+
+    private func removeSession(for url: WalletConnectURL) {
+        guard var sessions = _sessions.value else { return }
+
+        if let index = sessions.firstIndex(where: { $0.url.absoluteString == url.absoluteString }) {
+            set(server: nil, for: sessions[index].url)
+            sessions.remove(at: index)
+        }
+
+        UserDefaults.standard.walletConnectSessions = sessions
```

Comment:
Better to store a Config instance and read its `defaults` to have the correct `UserDefaults` instance?